src/sas/qtgui/Perspectives/Fitting/ModelThread.py

```diff
@@ -214,24 +214,37 @@ def compute(self):
         else:
             output[index] = self.model.evalDistribution(self.data.x[index])
 
-        sq_values = None
-        pq_values = None
-        s_model = None
-        p_model = None
-        if isinstance(self.model, MultiplicationModel):
-            s_model = self.model.s_model
-            p_model = self.model.p_model
-        elif hasattr(self.model, "calc_composition_models"):
-            results = self.model.calc_composition_models(self.data.x[index])
-            if results is not None:
-                pq_values, sq_values = results
+        results_eval = {}
+        intermediate_results = getattr(self.model, "_intermediate_results", None)
```

Comment:
I don't think sasmodels isn't yet saving the intermediate results in the model, even in the lazy branch.  

When it does, it should not store it with a variable name that starts with "_" since by convention that is a private variable subject to change, not a public part of the api to be used by code outside the class.

This interface still has threading issues.  If a separate thread evaluates the model it will override the immediate results saved within the model.  The interface within sasmodels could be changed so that calculate_Iq returns I(q) as well as the lazy results.  Sasview can be updated to call calculate_Iq directly, and get a handle directly to the lazy intermediate evaluator.